### ISSUE TITLE
RFC `find_good_eps`

### ISSUE THREAD

**Issue by xukai92** [TuringLang/AdvancedHMC.jl#27]
Copied from @yebai comment in https://github.com/TuringLang/AdvancedHMC.jl/pull/23#discussion_r268405153

```julia
    r = rand_momentum(rng, h)
    H = hamiltonian_energy(h, θ, r)
```

out of `find_good_eps`  and wraping

```julia
        θ′, r′, _is_valid = step(Leapfrog(ϵ′), h, θ′, r′)
        H_new = _is_valid ? hamiltonian_energy(h, θ′, r′) : Inf
```

into a  function 

```julia

function A(ϵ, h, θ, r) 
        θ′, r′, _is_valid = step(Leapfrog(ϵ′), h, θ′, r′)
        return H_new = _is_valid ? hamiltonian_energy(h, θ′, r′) : Inf
end
```

Then, we can drop the dependency on `AdvancedHMC` from this function: `find_good_eps(h::Hamiltonian, θ::AbstractVector{T}, A::Function; max_n_iters::Int=100)`, and move it into `adaption/stepsize.jl`.

**Comment by xukai92** [comment]
I refactored it anyway but I don't see any reason we have to drop the dependency any more. Plus it still depends on `Hamiltonian` any way?

**Comment by xukai92** [comment]
Closed by PR #79.
